### PR TITLE
readme: update individual ext readmes

### LINKED ISSUE #6518: [documentation] improve the documentation of extension 'readme'
**Description**

The `readme` is the first source of documentation of every extension when consumed from `npm`. I believe that we can definitely improve the documentation overall by updating the documentation present in the readme of extensions, describing to end-users and adopters the overall functionality the extension provides.

Some example of relevant information can include:
- overall description of the extension
- high-level features (widgets, functionality)
- contributed commands
- contributed preferences

My only worry is that if we do happen to update the documentation present in the `readme`, it might be a significant effort to keep updating (and also remember to update).

The documentation in the `readme` is however insufficient, for example:
https://www.npmjs.com/package/@theia/workspace

**Reason**
- informs users and adopters the high-level description of the extension
- informs users of relevant features and functionality the extension provides
- describes available commands
- describes available preferences

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes #6518

The following pull-request updates the `readme` for each individual extension following the updates made to the `getting-started` extension (#7077).

This includes:
- a uniform header (with theia icon, and extension name)
- a low maintenance description of the given extension
- helpful links including the generate API docs when possible

<details>
<summary>Updates</summary>

**dev-packages**

- [x] application-manager
- [x] application-package
- [x] cli
- [x] electron

**packages**

- [x] callhierarchy
- [x] console
- [x] core
- [x] debug
- [x] editor
- [x] editor-preview
- [x] file-search
- [x] filesystem
- [x] getting-started
- [x] git
- [x] json
- [x] keymaps
- [x] languages
- [x] markers
- [x] messages
- [x] metrics
- [x] mini-browser
- [x] monaco
- [x] navigator
- [x] outline-view
- [x] output
- [x] plugin
- [x] plugin-dev
- [x] plugin-ext
- [x] plugin-ext-vscode
- [x] plugin-metrics
- [x] preferences
- [x] preview
- [x] process
- [x] scm
- [x] scm-extra
- [x] search-in-workspace
- [x] task
- [x] terminal
- [x] typehierarchy
- [x] userstorage
- [x] variable-resolver
- [x] workspace

</details>

#### How to test

Verify the content of the `readmes`.

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: dev-packages/application-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<div align='center'>
+
+<br />
+
+<img src='https://raw.githubusercontent.com/eclipse-theia/theia/master/logo/theia.svg?sanitize=true' alt='theia-ext-logo' width='100px' />
+
+<h2>THEIA - APPLICATION-MANAGER EXTENSION</h2>
```
Comment (akosyakov) [inline dev-packages/application-manager/README.md]: it is not an extension, but a package providing API to build and start applications. The same for the application package and cli.

File: packages/core/README.md [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
+<h2>THEIA - CORE EXTENSION</h2>
+
+<hr />
+
+</div>
+
+## Description
+
+The `@theia/core` extension is the main extension which provides functionality, base widgets, and contribution points for all dependent extensions. At a high-level, the extension allows for all dependent extensions the ability to contribute additional functionality to the application. For example, an application can contribute commands, menu items, statusbar items, new dialogs, and new widgets all with the help of `@theia/core`.
```
Comment (akosyakov) [inline packages/core/README.md]: most importantly it provides application and shell APIs as well as contributions to hook into application lifecycles.

File: packages/plugin/README.md [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+
+<img src='https://raw.githubusercontent.com/eclipse-theia/theia/master/logo/theia.svg?sanitize=true' alt='theia-ext-logo' width='100px' />
+
+<h2>THEIA - PLUGIN EXTENSION</h2>
+
+<hr />
+
+</div>
+
+## Description
```
Comment (lmcbout) [inline packages/plugin/README.md]: Would be nice to put some description

File: dev-packages/application-manager/README.md (hunk 1 above)
Comment (vince-fugnitto) [reply dev-packages/application-manager/README.md]: > it is not an extension, but a package providing API to build and start applications. The same for the application package and cli.

Should I just drop `extension` as part of the name?

File: packages/core/README.md (hunk 2 above)
Comment (vince-fugnitto) [reply packages/core/README.md]: > most importantly it provides application and shell APIs as well as contributions to hook into application lifecycles.

Thank you, I wanted some help to describe `@theia/core` (there was no formal description before) in a way that would help extension developers.

File: packages/plugin/README.md (hunk 3 above)
Comment (vince-fugnitto) [reply packages/plugin/README.md]: @lmcbout I thought the description would be broken down into the content that was already present, please see the rendered version:
https://github.com/eclipse-theia/theia/blob/57cf060e2dd43370dd5ac5b27c828d17448ac553/packages/plugin/README.md

**Comment by akosyakov** [comment]
We can improve later. Let's merge?

**Comment by vince-fugnitto** [comment]
> We can improve later. Let's merge?

Sounds good, I'll merge and we can improve iteratively.

**Pull request merged**